plutus-contract/test/Spec/TxConstraints/MustMint.hs

```diff
@@ -260,14 +321,31 @@ mkMustMintPolicy t = case t of
     MustMintValueWithRedeemer r v           -> Constraints.checkScriptContext @() @() (Constraints.mustMintValueWithRedeemer r v)
     MustMintValue v                         -> Constraints.checkScriptContext @() @() (Constraints.mustMintValue v)
 
+{-# INLINEABLE mkMustMintPolicyV2 #-}
+mkMustMintPolicyV2 :: ConstraintParams -> PV2.ScriptContext -> Bool
+mkMustMintPolicyV2 t = case t of
+    MustMintCurrencyWithRedeemer mph r tn i -> TCV2.checkScriptContext @() @() (Constraints.mustMintCurrencyWithRedeemer mph r tn i)
+    MustMintCurrency mph tn i               -> TCV2.checkScriptContext @() @() (Constraints.mustMintCurrency mph tn i)
+    MustMintValueWithRedeemer r v           -> TCV2.checkScriptContext @() @() (Constraints.mustMintValueWithRedeemer r v)
+    MustMintValue v                         -> TCV2.checkScriptContext @() @() (Constraints.mustMintValue v)
```

Comment:
Will want to add the new `ConstraintParams` to use in the policy:
- mustMintCurrencyWithRedeemerAndReference
- mustMintValueWithRedeemerAndReference
- mustMintValueWithReference
- mustMintCurrencyWithReference

Then you want to use one for your positive case test `mustReferenceOutputV2Validator` and for one test demonstrating phase-2 failure due to unexepcted reference txo.